scale/job/views.py

```diff
@@ -699,18 +699,25 @@ def retrieve(self, request, job_exe_id, log_id):
         elif log_id == "stderr":
             include_stdout = False
 
+        started = rest_util.parse_timestamp(request, 'started', required=False)
         since = request.META.get('HTTP_IF_MODIFIED_SINCE', '')
         if len(since) > 0:
             since = util.parse.datetime.datetime.strptime(since, "%a, %d %b %Y %X %Z")
         else:
             since = None
 
+        if since is not None and started != since:
+            logs, last_modified = job_exec.get_log_json(include_stdout, include_stderr, since)
+            if logs is None:
+                rsp = HttpResponse(status=304)
+                rsp["Last-Modified"] = last_modified.strftime("%a, %d %b %Y %X %Z")
+                return rsp
```

Comment:
Having the two different timestamps (since, started) seems overly complicated and requires an extra ES API call. I would recommend that we just don't use IF_MODIFIED_SINCE.